pkg/validators/test_am0004.go

```diff
@@ -0,0 +1,52 @@
+package validators
+
+import (
+	"github.com/mt-sre/addon-metadata-operator/api/v1alpha1"
+	"github.com/mt-sre/addon-metadata-operator/internal/testutils"
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+)
+
+func init() {
+	TestRegistry.Add(TestAM0004{})
+}
+
+type TestAM0004 struct{}
+
+func (val TestAM0004) Name() string {
+	return AM0004.Name
+}
+
+func (val TestAM0004) Run(mb utils.MetaBundle) (bool, string, error) {
+	return AM0004.Runner(mb)
+}
+
+func (val TestAM0004) SucceedingCandidates() []utils.MetaBundle {
+	defaultSucceedingCandidates := testutils.DefaultSucceedingCandidates()
```

Comment:
nit: rename `defaultSucceedingCandidates` to `res` (just personal preference to name return value `res`)